python/lsst/ts/wep/donutSizeCorrelator.py

```diff
@@ -0,0 +1,344 @@
+        image -= np.median(image)
+
+        # Get bad-pixel mask
+        bits = exposure.mask.getPlaneBitMask(badPixelTypes)
+        badPixels = np.bitwise_and(exposure.mask.array, bits) > 0
+
+        # Dilate the bad-pixel mask
+        mask = binary_dilation(badPixels, iterations=nDilation)
+
+        # Fill masked values
+        image[mask] = fillVal
+
+        # Normalize the image
+        image /= image[~mask].max()
+
+        return image
+
+    @staticmethod
+    def cropAndBinImage(
+        image: np.ndarray,
+        length: int | None = None,
+        pad: int = 500,
+        binning: int | None = None,
+    ) -> np.ndarray:
+        """Crop and bin the array.
+
+        Parameters
+        ----------
+        image : np.ndarray
+            Image array.
+        length : int or None, optional
+            Size length for crop. Default is None.
+        pad : int, optional
+            The image is cropped around the brightest pixel, subject
+            to the condition that the pixel is not too close to the
+            edge of the image. This pad sets that distance.
+            Default is 500 pixels.
+        binning : int or None, optional
+            Binning factor. Default is None.
+
+        Returns
+        -------
+        np.ndarray
+            Cropped and binned image
+        """
+        # Crop array
+        if length is not None:
+            # We will crop around the brightest pixel
+
+            # Cutout pads on side so we don't select for
+            # bright donuts falling off the sensor
+            padded = image[pad:-pad, pad:-pad]
+
+            # Bin the array so we don't select individual hot pixels
+            binning_factor = 8
+            binned = binArray(padded, binning_factor)
+
+            # Find location of brightest pixel
+            y, x = np.where(binned == np.nanmax(binned))
+
+            # Undo binning and pad
+            x = binning_factor * x[0] + pad
+            y = binning_factor * y[0] + pad
+
+            # Crop around brightest pixel
+            H, W = image.shape
+            x0 = np.clip(x - length // 2, 0, W - length)
+            y0 = np.clip(y - length // 2, 0, H - length)
+            image = image[y0 : y0 + length, x0 : x0 + length]
+
+        # Bin array using median to avoid impact of hot pixels
+        if binning is not None:
+            image = binArray(image, binning, "median")
+
+        return image
+
+    @staticmethod
+    def createDonutTemplate(diameter: float) -> np.ndarray:
+        """Create simple annulus template for donuts.
+
+        Parameters
+        ----------
+        diameter : float
+            Diameter of donut.
+
+        Returns
+        -------
+        np.ndarray
+            Fractional donut mask
+        """
+        # Create grid of pixel centers
+        x = np.arange(diameter + 5, dtype=float)
+        x -= x.mean()
+        x, y = np.meshgrid(x, x)
+
+        # Build sub-pixel offsets
+        N_offsets = 10
```

Comment:
The variable names in this class are a mix of snake case and camel case. It needs to be consistent. In `ts_wep` we have used camel case in our other classes.

Reply:
See also `d_min`, `d_max` below